dpppt-backend-sdk/dpppt-backend-sdk-data/src/main/java/org/dpppt/backend/sdk/data/gaen/JDBCGAENDataServiceImpl.java

```diff
@@ -127,6 +133,13 @@ public int getMaxExposedIdForKeyDate(
             + " and rolling_start_number < :rollingPeriodStartNumberEnd and received_at <"
             + " :publishedUntil";
 
+    if (now != null) {
+      params.addValue(
+          "maxAllowedStartNumber",
+          now.roundToPreviousBucket(releaseBucketDuration).plusHours(2).get10MinutesSince1970());
+      sql += " and rolling_start_number + rolling_period < :maxAllowedStartNumber";
+    }
+
     if (publishedAfter != null) {
       params.addValue("publishedAfter", publishedAfter.getDate());
       sql += " and received_at >= :publishedAfter";
```

Comment:
Possibly add a comment that `received_at` is rounded to (almost) correspond to the batch boundaries, so this is correct.

Reply:
Added some more comments in the `getSortedExposedForKeyDate` function. Is it now clearer?